### PR TITLE
Add documentation for rebasing autoscaler and revendoring MCM in FAQ

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**: Adding documentation for rebasing autoscaler and revendoring MCM in the FAQ.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Comment by prashanth26** [comment]
Thanks for this PR @hardikdr . Might need a day or two to test this change out.

File: cluster-autoscaler/FAQ.md [hunk 1]
```diff
@@ -953,3 +957,190 @@ Caveats:
...
+cd autoscaler/cluster-autoscaler
+git remote add origin git@github.com:gardener/autoscaler.git
+
+git checkout machine-controller-manager-provider
+git pull origin machine-controller-manager-provider
+
+git log | grep "Updating vendor against"  # Please save commit-hash from the first line of the output, it'll be used later, eg: 3eb90c19d0cf90b756c3e08e32c6495b91e0aeed
+
+``` 
+
```
Comment (prashanth26) [inline cluster-autoscaler/FAQ.md]: Step 3 should probably be moved above step 2?

File: cluster-autoscaler/FAQ.md [hunk 2]
```diff
@@ -953,3 +957,190 @@ Caveats:
...
+#### Step 4:
+
+This fork of the autoscaler vendors the [machine-controller-manager](https://github.com/gardener/machine-controller-manager) aka MCM from Gardener project. As the MCM itself vendors the `k8s.io` in it, we need to make following change to the [`update-vendor`](https://github.com/gardener/autoscaler/blob/master/cluster-autoscaler/hack/update-vendor.sh) script:
+
+Disable the check of implicit-depedencies of go.mod by commenting out following code in the update-vendor script.
+
+```
+#  if [[ "${IMPLICIT_FOUND}" == "true" ]]; then
+#    err_rerun "Implicit dependencies missing from go.mod-extra"
+#  fi
```
Comment (prashanth26) [inline cluster-autoscaler/FAQ.md:1112]: Do we require this? I was able to vendor without this. Or maybe we can move it as optional step if you feel it's required at times.

File: cluster-autoscaler/FAQ.md [hunk 3]
```diff
@@ -953,3 +957,190 @@ Caveats:
...
+```
+
+Populate the `K8S_REV` variable in the script with the commit-hash you saved above, as `-r` flag of the original-script in the command-line doesn't work for few environments. Eg.
```
Comment (prashanth26) [inline cluster-autoscaler/FAQ.md:1115]:
```suggestion
Populate the `K8S_REV` variable in the script with the commit-hash you saved above. 
Also `-r` flag used by `xargs` command of the original-script in the command-line doesn't work for few environments, remove this flag out before running in these environments.

`xargs -t -r go mod edit` --> `xargs -t go mod edit`
```

**Comment by prashanth26** [comment]
Thanks for this documentation. I just have a few minor comments. /lgtm otherwise.

File: cluster-autoscaler/FAQ.md (hunk 1 above)
Comment (hardikdr) [reply cluster-autoscaler/FAQ.md]: Sure, sounds good. Done.

File: cluster-autoscaler/FAQ.md (hunk 2 above)
Comment (hardikdr) [reply cluster-autoscaler/FAQ.md:1112]: I made it `Optional` now.

File: cluster-autoscaler/FAQ.md (hunk 3 above)
Comment (hardikdr) [reply cluster-autoscaler/FAQ.md:1115]: I am actually referring to this flag: https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/hack/update-vendor.sh#L33 of the script, and not of `xargs`.
- Issue with `xargs` seems to have been fixed upstream already.

**Review by prashanth26 — APPROVED** [review]
@hardikdr - Feel free to merge when you feel it's ready.
/lgtm

**Pull request merged**